extensions/interactions/SetInput/directives/oppia-interactive-set-input.component.ts

```diff
@@ -0,0 +1,142 @@
+// Copyright 2014 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Component for the SetInput interaction.
+ *
+ * IMPORTANT NOTE: The naming convention for customization args that are passed
+ * into the component is: the name of the parameter, followed by 'With',
+ * followed by the name of the arg.
+ */
+
+import { Component, Input, OnInit } from '@angular/core';
+import { InteractionAttributesExtractorService } from 'interactions/interaction-attributes-extractor.service';
+import { CurrentInteractionService } from 'pages/exploration-player-page/services/current-interaction.service';
+import { SetInputRulesService } from './set-input-rules.service';
+import eq from 'lodash/eq';
+import { InteractionRulesService } from 'pages/exploration-player-page/services/answer-classification.service';
+import { SetInputCustomizationArgs } from 'interactions/customization-args-defs';
+import { Schema } from 'services/schema-default-value.service';
+import { downgradeComponent } from '@angular/upgrade/static';
+
+@Component({
+  selector: 'oppia-interactive-set-input',
+  templateUrl: './set-input-interaction.component.html'
+})
+export class InteractiveSetInputComponent implements OnInit {
+  @Input() buttonTextWithValue: string;
+  @Input() savedSolution;
+  errorMessage: string;
+  answer;
+  schema: {
+    type: string;
+    items: {
+      type: string;
+    };
+    'ui_config': {
+      'add_element_text': string;
+    };
+  };
```

Comment:
Might also make sense to separate this into a type variable.

Reply:
Will do it in a followup PR